src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -1060,6 +1118,39 @@ def upgrade_agents(cmd, client, resource_group_name, cluster_name, kube_config=N
                                 summary='Problem loading the helm existing user supplied values')
         raise CLIInternalError("Problem loading the helm existing user supplied values: " + str(e))
 
+    # Change registry_path to private registry if PCR is set in helm values
+    existing_user_values_flat = utils.flatten(existing_user_values)
+    registry_username = None
+    registry_password = None
+    isContainerRegistryEnabled = existing_user_values_flat.get('global.isCustomRegistryEnabled')
+
+    if isContainerRegistryEnabled and arc_agent_version is None:
+        logger.warning("Skipping upgrade as Arc agent version is not provided for custom container registry")
```

Comment:
Since, it s user fault ideally for PCR scenarion, Let's raise ArgumentUsageError("Please provide agent version for upgrading the arc agents on the kubernetes cluster.", recommendation="use --agent-version")

also add telemetry exception.

Reply:
ack